### PR TITLE
feat: example groupby Document

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

**Review by theUnkownName — APPROVED** [review]

**Comment by hanxiao** [comment]
please change according to my comments

**Review by hanxiao — CHANGES_REQUESTED** [review]
change required

File: .github/2.0/cookbooks/Document.md [hunk 1]
```diff
@@ -576,6 +576,17 @@ DocumentArray has 3 items:
...
 As `DocumenArray` is an `Iterable`, you can also use [Python built-in `itertools` module](https://docs.python.org/3/library/itertools.html) on it. This enables advanced "iterator algebra" on the `DocumentArray`.
 
+For instance, you can group a `DocumentArray` by parent_id:
+```python
+from jina import DocumentArray, Document
+from itertools import groupby
+
+da = DocumentArray()
+da.extend([Document(parent_id=f'{i%2}') for i in range(6)])
+groups = groupby(sorted(da, key=lambda d: d.parent_id), lambda d: d.parent_id)
```
Comment (hanxiao) [inline .github/2.0/cookbooks/Document.md:586]: can we remove sorted? it is not necessary right?

**Review by hanxiao — CHANGES_REQUESTED** [review]
remove sorted

File: .github/2.0/cookbooks/Document.md (hunk 1 above)
Comment (JoanFM) [reply .github/2.0/cookbooks/Document.md:586]: it is, groupby groups consecutive entries

Comment (JoanFM) [reply .github/2.0/cookbooks/Document.md:585]: If I remove sorted I cannot interlace different parent_ids. And most of the time it will be needed

**Pull request merged**